### PR TITLE
Fix issue #183 image.itemtype

### LINKED ISSUE #183: Warning: The attribute image.itemtype has an invalid value.
Looks like when I removed all of my image changes in #151 (because of #174), I also removed my change that appeased Google's "Warning" `The attribute image.itemtype has an invalid value.`. 

We have:
```
"image": "https://www.kyleniewiada.org/assets/img/2017/02/banner.png",
```

We should have:
```
"image": {"@type": "ImageObject",
"url": "https://www.kyleniewiada.org/assets/img/2017/02/banner.png",
"height": 400,
"width": 700},
```

Hence why we offer the image `height` and `width` as options. I can add this back in, but it won't be until the weekend. It's somewhere in one of the older commits for #151, along with some tests.

### PR DESCRIPTION
**Pull request by aav7fl**
- [X]  Add Feature
- [X]  Add Tests

Fixes #183 by adding back the image.itemtype to the JSON-LD to pass the [Google Structured Data Testing](https://search.google.com/structured-data/testing-tool/u/0/) tool.

I've added the JSON-LD tests within the same context of the (already present) image tests as it made sense from a refactor standpoint.. let me know if you'd like that pulled to a different level.

### CODE REVIEW

File: lib/template.html [hunk 1]
```diff
@@ -223,7 +223,16 @@
 {% endif %}
 
 {% if seo_page_image %}
+  {% if page.image.height && page.image.width %}
+    "image": {
+        "@type": "ImageObject",
+        "url": {{ seo_page_image | jsonify }},
```
Comment (benbalter) [inline lib/template.html]: Based on https://developers.google.com/schemas/reference/types/MediaObject should this be `contentUrl`?

Comment (aav7fl) [inline lib/template.html]: Google doesn't like that.
http://i.imgur.com/RtcpMeQ.png

Comment (benbalter) [inline lib/template.html]: Odd. 🤷‍♂️   I got there via https://developers.google.com/schemas/reference/types/ImageObject.

Comment (aav7fl) [inline lib/template.html]: My guess is that the JSON-LD checker is using this: https://developers.google.com/search/docs/data-types/articles (See `image` / `ImageObject`)

**Comment by benbalter** [comment]
:metal: Thanks @aav7fl for the PR and for talking through it with me. Going to merge this and then reconcile the changes for https://github.com/jekyll/jekyll-seo-tag/pull/184.

**Comment by aav7fl** [comment]
Not a problem. I'm really looking forward to the drop. It looks really good!

**Comment by benbalter** [comment]
> Going to merge this and then reconcile the changes for #184.

Done via bc0f45a

**Pull request merged**